File: lambda-kinesis-lambda/consumer/consumer-function.py
import base64

print('Loading function')

def lambda_handler(event, context):
    #print("Received event: " + json.dumps(event, indent=2))
    for record in event['Records']:
        # Kinesis data is base64 encoded so decode here
        data = base64.b64decode(record['kinesis']['data']).decode('utf-8')
        print("Decoded data: " + data)
    return 'Successfully processed {} records.'.format(len(event['Records']))

File: lambda-kinesis-lambda-js/consumer/consumer-function.js
//! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0
exports.lambda_handler = async (event, context) => {
    const records = event.Records;

    for (const record of records) {
        const payload = Buffer.from(record.kinesis.data, 'base64').toString('utf-8');
        console.log('Received payload:', payload);
        // Add your custom logic to process the payload here
    }

    return { statusCode: 200, body: 'Success' };
};